### PR TITLE
feat(primitive): add text component

### PR DESCRIPTION
**Pull request by corquaid**
refer issue #74

Description
---
Add generic Text component

Motivation and Context
---
Primitive Text component connected to fonts/typography

How Has This Been Tested?
---
Unit tests will be added in future PR when testing structure is set up.

Screenshot 2022-04-20 at 10 11 05

### CODE REVIEW

File: applications/launchpad_v2/src/components/Text/index.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  color,
+  children,
+}: TextProps) => {
+  const textStyles = {
+    color: color,
+    ...styles.typography[type]
+  }
+  return (
+    <>
+      <GlobalFonts />
```
Comment (tarnas14) [inline applications/launchpad_v2/src/components/Text/index.tsx]: global styles should be included once per application (as they're global)
imo this should be rendered inside <App /> component

**Review by tarnas14 — CHANGES_REQUESTED** [review]
see comment about global styles

**Comment by tarnas14** [comment]
just a question - so the idea is that whenever I'm rendering text (e.g. when I'm showing a label above a select component), I should use <Text> component, right?

if so, I think `type` should be optional and have a nice default value (I guess `defaultMedium` ? )

File: applications/launchpad_v2/src/components/Text/index.tsx (hunk 1 above)
Comment (corquaid) [reply applications/launchpad_v2/src/components/Text/index.tsx]: Fixed, global styles are now in App.tsx

File: applications/launchpad_v2/src/App.tsx [hunk 2]
```diff
@@ -18,26 +19,28 @@ function App() {
...
-        <header className="App-header">
-          <img src={logo} className="App-logo" alt="logo" />
-          <p>
+      <GlobalStyle />
+      <>
+        <div className="App">
+          <header className="App-header">
+            <img src={logo} className="App-logo" alt="logo" />
+            <p>
           Edit <code>src/App.tsx</code> and save to reload.
```
Comment (tarnas14) [inline applications/launchpad_v2/src/App.tsx]: this indentation is weird, why does our linter not complain about this?

File: applications/launchpad_v2/src/App.tsx [hunk 3]
```diff
@@ -18,26 +19,28 @@ function App() {
 
   return (
     <ThemeProvider theme={{}}>
-      <div className="App">
-        <header className="App-header">
-          <img src={logo} className="App-logo" alt="logo" />
-          <p>
+      <GlobalStyle />
+      <>
```
Comment (tarnas14) [inline applications/launchpad_v2/src/App.tsx]: you're wrapping single component inside a fragment, either GlobalStyle should be inside here to satisfy ThemeProvider children props or the fragment is not needed

File: applications/launchpad_v2/src/styles/styles/index.ts [hunk 4]
```diff
@@ -1,6 +1,6 @@
 import colors from './colors'
 import gradients from './gradients'
-import typography from '../typography'
+import  typography from './typography'
```
Comment (tarnas14) [inline applications/launchpad_v2/src/styles/styles/index.ts]: again, why doesn't linter not complain about this additional space between `import` and `typography` ?

**Review by tarnas14 — DISMISSED** [review]
exactly what we discussed, minor comments about linting and fragments, otherwise looks good to go after fragment change

File: applications/launchpad_v2/src/App.tsx (hunk 3 above)
Comment (corquaid) [reply applications/launchpad_v2/src/App.tsx]: Fixed, my error.

File: applications/launchpad_v2/src/styles/styles/index.ts (hunk 4 above)
Comment (corquaid) [reply applications/launchpad_v2/src/styles/styles/index.ts]: not sure on the linter checks.

Comment (tarnas14) [reply applications/launchpad_v2/src/styles/styles/index.ts]: rip

**Review by tarnas14 — DISMISSED** [review]
ship it

**Review by tomaszantas — APPROVED** [review]
LGTM!

**Pull request merged**